### DISCUSSION TITLE
Is it safe to instantiate a resource more then once?

### DISCUSSION THREAD

**Discussion by liborjelinek** [falconry/falcon#2158] [Q&A] [answered]
Looking at the example in [suffixed response](https://falcon.readthedocs.io/en/stable/api/routing.html#suffixed-responders), when the above question arose.

What is the difference between (one instance):

```
calc = Calculator()
app = falcon.App()
app.add_route('/add', calc, suffix='add')
app.add_route('/subtract', calc, suffix='subtract')
```

and two-times instantiated resource:

```
app = falcon.App()
app.add_route('/add', Calculator(), suffix='add')
app.add_route('/subtract', Calculator(), suffix='subtract')
```

?

**Comment by vytas7** [comment, marked as answer]
Hi again @mattwarrick!
Yes, it is absolutely up to you, the developer, whether it is safe or not.

AFAIK, Falcon doesn't make any assumptions about resource instances apart from (when using the default router) just looking for callable attributes that look like responders (e.g., `on_get`, `on_post`, etc). I suppose you might have some (probably negligible) memory savings reusing the same instance, but the framework doesn't mandate it.

Although it is probably a bit strange to instantiate `Calculator()` again in the above example, but in the case a resource's initializer takes parameters, it might be even desirable to create different resources instances from the same class, for instance:
```python
class Resource:
    def __init__(self, database_pool, ...):
        ...

app.add_route('/cache/objects/{object_id}', Resource(ro_db_replica_pool))
app.add_route('/objects/{object_id}', Resource(rw_db_pool))
```
